src/transformers/models/lightglue/modular_lightglue.py

```diff
@@ -138,9 +146,24 @@ def __init__(
             keypoint_detector_config["model_type"] = (
                 keypoint_detector_config["model_type"] if "model_type" in keypoint_detector_config else "superpoint"
             )
-            keypoint_detector_config = CONFIG_MAPPING[keypoint_detector_config["model_type"]](
-                **keypoint_detector_config, attn_implementation="eager"
-            )
+            if keypoint_detector_config["model_type"] not in CONFIG_MAPPING and self.trust_remote_code:
+                AutoConfig.register(
+                    keypoint_detector_config["model_type"],
+                    AutoConfig.from_pretrained(
+                        keypoint_detector_config["_name_or_path"], trust_remote_code=self.trust_remote_code
+                    ).__class__,
+                )
+            keypoint_detector_config_class = CONFIG_MAPPING[keypoint_detector_config["model_type"]]
+
+            if self.trust_remote_code:
+                keypoint_detector_config = keypoint_detector_config_class(
+                    **keypoint_detector_config, attn_implementation="eager", trust_remote_code=self.trust_remote_code
+                )
+            else:
+                keypoint_detector_config = keypoint_detector_config_class(
+                    **keypoint_detector_config, attn_implementation="eager"
+                )
```

Comment:
I think that's the only change we need in this file, would it work?

```suggestion
            trust_remote_code = kwargs.get("trust_remote_code", None)
            if keypoint_detector_config["model_type"] not in CONFIG_MAPPING:
                    keypoint_detector_config = AutoConfig.from_pretrained(
                        keypoint_detector_config["_name_or_path"], trust_remote_code=trust_remote_code
                    )
            else:
                keypoint_detector_config = keypoint_detector_config_class(
                    **keypoint_detector_config, attn_implementation="eager"
                )
```

Reply:
As I mentioned in the OP, I was surprised that `trust_remote_code` is not included in the kwargs when using `from_pretrained`, it is popped at this line : 
https://github.com/huggingface/transformers/blob/838a0268b88b6e56783b401f9cacae9cb0cbb120/src/transformers/modeling_utils.py#L4449

You can try it with the following snippet : 
```python
from transformers import LightGlueForKeypointMatching
model = LightGlueForKeypointMatching.from_pretrained("ETH-CVG/lightglue_disk", trust_remote_code=True)
```

That's the only workaround I found to make it work